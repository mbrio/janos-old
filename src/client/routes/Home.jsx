import React from 'react';
import Helmet from 'react-helmet';
import autobind from 'autobind-decorator';
import WindowSizedComponent from '../containers/WindowSizedComponent';

/**
 * The home route, is composed with WindowSizedComponent so that it understands when the window is
 * resized.
 */
@WindowSizedComponent
export default class Home extends React.Component {
  static propTypes = {
    windowWidth: React.PropTypes.number.isRequired,
    windowHeight: React.PropTypes.number.isRequired,
  }

  state = {
    message: 'Hello, App!',
  }

  /**
   * The button click event handler, it is defined as an arrow function so that it is bound to the
   * instance of the component.
   *
   * @param {Object} e - The event object passed in by the browser.
   */
  @autobind
  handleClick(e) {
    e.preventDefault();

    this.setState({ message: Math.random() });
  }

  render() {
    return (
      <div>
        <Helmet title="Home" />
        <div>{this.state.message}</div>
        <div>{this.props.windowWidth}x{this.props.windowHeight}</div>
        <div><button onClick={this.handleClick}>Click Me</button></div>
      </div>
    );
  }
}
