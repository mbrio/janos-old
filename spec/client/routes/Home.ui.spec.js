import React from 'react';
import Home from '../../../src/client/routes/Home.jsx';
import renderer from 'react-test-renderer';

describe('Home', () => {
  describe('UI', () => {
    it('renders with expected output', () => {
      const component = renderer.create(
        <Home windowWidth="1000" windowHeight="1000" />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
