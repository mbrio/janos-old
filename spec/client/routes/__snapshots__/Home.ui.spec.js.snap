exports[`Home UI renders with expected output 1`] = `
<div>
  <div
    className="label">
    Hello, App!
  </div>
  <div>
    <button
      onClick={[Function bound handleClick]}>
      Click Me
    </button>
  </div>
</div>
`;
